components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/authcontext/DefaultClaimsRetrieverTest.java

```diff
@@ -0,0 +1,242 @@
+        userClaimUri1 = "testUserClaimURI1";
+        claim3.setClaimUri(userClaimUri1);
+        testMappedAttributesUser1 = "MappedAttributesUser1";
+
+        userClaimUri2 = "testUserClaimURI2";
+        claim4.setClaimUri(userClaimUri2);
+        testMappedAttributesUser2 = "MappedAttributesUser2";
+
+        ClaimMapping cMap1 = new ClaimMapping(claim1, testMappedAttributesCore1);
+        ClaimMapping cMap2 = new ClaimMapping(claim2, testMappedAttributesCore2);
+        ClaimMapping cMap3 = new ClaimMapping(claim3, testMappedAttributesUser1);
+        ClaimMapping cMap4 = new ClaimMapping(claim4, testMappedAttributesUser2);
+
+        ClaimMapping[] coreClaims = new ClaimMapping[]{cMap1, cMap2, cMap3, cMap4};
+
+
+        return coreClaims;
+    }
+
+    @Test
+    public void testInit() throws Exception {
+        // Subject is null.
+        assertFalse(Boolean.parseBoolean(defaultClaimsRetriever.DEFAULT_DIALECT_URI));
+
+        // Subject is not null.
+        mockStatic(OAuthServerConfiguration.class);
+        when(OAuthServerConfiguration.getInstance()).thenReturn(mockedOAuthServerConfiguration);
+        when(mockedOAuthServerConfiguration.getConsumerDialectURI()).thenReturn("consumerDialectURI");
+        assertFalse(Boolean.parseBoolean(mockedOAuthServerConfiguration.getConsumerDialectURI()));
+    }
+
+    @Test
+    public void testGetClaims() throws Exception {
+        SortedMap<String, String> claimValues;
+        int tenantId = MultitenantConstants.SUPER_TENANT_ID;
+
+        mockStatic(OAuthServerConfiguration.class);
+        when(OAuthServerConfiguration.getInstance()).thenReturn(mockedOAuthServerConfiguration);
+        when(mockedOAuthServerConfiguration.getTimeStampSkewInSeconds()).thenReturn((long) 3);
+        mockStatic(OAuth2Util.class);
+        mockStatic(MultitenantUtils.class);
+        mockStatic(OAuthComponentServiceHolder.class);
+
+        when(OAuthComponentServiceHolder.getInstance()).thenReturn(mockedOAuthComponentServiceHolder);
+        when(mockedOAuthComponentServiceHolder.getRealmService()).thenReturn(mockedRealmService);
+        mockedUserRealm = mock(UserRealm.class);
+        when(mockedRealmService.getTenantUserRealm(anyInt())).thenReturn(mockedUserRealm);
+        when(mockedUserRealm.getUserStoreManager()).thenReturn(mockedUserStoreManager);
+        Map<String, String> expectedMappingTrue = new HashMap<>();
+        when(mockedUserStoreManager.getUserClaimValue("user", "[\"https://www.wso2.org/address\", \"https://www.wso2.org/email\"]", "read")).thenReturn(expectedMappingTrue.toString());
+
+        String[] claims = {"https://www.wso2.org/address", "https://www.wso2.org/email"};
+
+        assertNotNull(defaultClaimsRetriever.getClaims("admin", claims));
+    }
+
+    @Test(expectedExceptions = IdentityOAuth2Exception.class)
+    public void testGetClaims1() throws Exception {
+        SortedMap<String, String> claimValues;
+        int tenantId = MultitenantConstants.SUPER_TENANT_ID;
+
```

Comment:
We haven't used this variable anywhere within this method. If it is not used remove it.

Reply:
Done